### PR TITLE
CMR-7180: add checking for collection-specific granule OSDD and use that link before using tagging method

### PR DESCRIPTION
**Pull request by charleshuang80**

### CODE REVIEW

File: app/models/collection.rb [hunk 1]
```diff
@@ -121,7 +122,10 @@ def to_open_search_collections(document, hits, params, url, for_atom = true, cwi
           add_link_as_child(doc, node, href, 'text/html', 'enclosure', short_name)
         end
 
-        if provider_osdd_link.present?
```
Comment (littleidiot40) [inline app/models/collection.rb:124]: I believe the form of a provider OSDD link has changed (https://bugs.earthdata.nasa.gov/browse/ECSE-995) does that require a change in the present? method or does https://bugs.earthdata.nasa.gov/browse/CMR-7178 abstract that away from you?

Comment (charleshuang80) [reply app/models/collection.rb:124]: The provider_osdd_link is still using the tags to identify it, but this `collection_specific_osdd_link` is using the new rel value of `'http://esipfed.org/ns/fedsearch/1.1/search#'` to identify it. And if I understood the change required correctly, we are looking first for the new link type first and using that if it exists, and if not they we will look for the older tag identified link to provide. Did I misunderstand anything?

**Review by eereiter — APPROVED** [review]

File: app/models/collection.rb [hunk 2]
```diff
@@ -208,6 +213,26 @@ def provider_granule_osdd_collection_link(entry_node)
...
   end
 
+  # a collection-specific granule OSDD would be provided as a Related URL
+  # that is a link node with a specific rel
+  def collection_specific_granule_osdd_link(entry_node, collection_granule_osdd)
+    if collection_granule_osdd.nil?
+      # not yet assigned
+      if entry_node.name == 'link' && entry_node.attributes.dig('rel')&.value == 'http://esipfed.org/ns/fedsearch/1.1/search#'
+        collection_granule_osdd = entry_node.attributes.dig('href')&.value
+        puts "assigned collection_granule_osdd: #{collection_granule_osdd}"
```
[the remark this answers is not in the document]
Comment (littleidiot40) [reply app/models/collection.rb:224]: Remove puts and/or add logging.

**Review by littleidiot40 — CHANGES_REQUESTED** [review]

File: app/models/collection.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (charleshuang80) [reply app/models/collection.rb]: missed this somehow. removed now.

**Review by littleidiot40 — APPROVED** [review]

**Pull request merged**